processor/lib/partitionPump.ts

```diff
@@ -200,4 +207,21 @@ export class PartitionPump {
       log.partitionPump(withHostAndPartition(partitionId, "No receiver was found to remove."));
     }
   }
+
+  private _isReceiverDisconnectedError(error: MessagingError | Error): boolean {
+    const partitionId = this._partitionContext.partitionId;
+    const withHostAndPartition = this._context.withHostAndPartition;
+    let result = false;
+    if (error) {
+      // condition is "amqp:link:stolen"
+      if ((error as MessagingError).condition === ErrorNameConditionMapper.ReceiverDisconnectedError) {
+        result = true;
+      } else if (error.message.match(/.*New receiver with higher epoch.*/i) !== null) {
+        result = true;
+        log.error(withHostAndPartition(partitionId, "It looks like the error should have " +
+          "been a 'ReceiverDisconnectedError', however it was not translated correctly: %O."), error);
+      }
+    }
+    return result;
```

Comment:
>New receiver with higher epoch of '5' is created hence current receiver with epoch '4' is getting disconnected. If you are recreating the receiver, make sure a higher epoch is used. TrackingId:ff92ddfe000003c30004c7925ba3a318_G9_B20, SystemTracker:streams:eventhub:test-hub~24575|$default, Timestamp:9/20/2018 1:39:41 PM

This will ensure that such errors are not bubbled up to the user as reported in #159